### ISSUE TITLE
`PrescribedAtmosphere` has redundant `runoff_flux` and `freshwater_flux`

### ISSUE THREAD

**Issue by glwagner** [CliMA/ClimaOcean.jl#131]
`PrescribedAtmosphere` has two types who's purpose appear to overlap:

https://github.com/CliMA/ClimaOcean.jl/blob/e2b8280bd8ac21c5885fb65056464f5e77612a63/src/OceanSeaIceModels/PrescribedAtmospheres.jl#L295-L296

because `runoff_flux` (presumably shorthand for "freshwater_river_flux" is also a `freshwater_flux`. 

Is runoff flux left out of "freshwater_flux"? If so we should use different names, because the river flux is also a freshwater flux.

**Comment by glwagner** [comment]
I think what is really need here is the capacity to have freshwater flux products that use a different grid and times and the rest. I don't think this is a specific concern to "runoff", it concerns any flux at all and there is no need to label these "runoff" (I'm also confused whether icebergs actually count as "runoff", this seems wrong).

Perhaps the new type should be called `auxiliary_freshwater_flux` or `secondary_freshwater_flux` which can be defined as flux components that have a different grid than the primary components (their organization largely being for performance it would seem, since the main point is that we want to avoid using `GPUAdaptedFieldTimeSeries`? I'm also confused about the motivation here and why we don't just use `GPUAdaptedFieldTimeSeries`.)

**Comment by simone-silvestri** [comment]
I think one of the main concerns here is to try to keep parameter space at a minimum. We can try and see if `GPUAdaptedFieldTimeSeries` works, but I am a little afraid that grids consume a lot of parameter space

**Comment by glwagner** [comment]
Right, so to solve that we would have to perform the interpolation separately (perhaps each field time series getting it's own kernel).

Although have you checked whether the compiler can detected aliased kernel args? For example, if we pass the same grid twice, we may only pay once for parameter space.

**Comment by simone-silvestri** [comment]
Hmmm right. 

By the way, this parameter space issue, as well as atmospheric fields being on different grids, may motivate another strategy, interpolating the atmospheric fields on the simulation grid when setting up the FTS memory in a similar way we do for ecco fields. This is done when we have to update the fields in memory
https://github.com/CliMA/ClimaOcean.jl/blob/e2b8280bd8ac21c5885fb65056464f5e77612a63/src/DataWrangling/ecco_restoring.jl#L44-L66

In this way, we could have all the atmospheric fields always on the correct grid and avoid passing too many arguments or having different atmospheric properties on different grids. The downside of this approach is that for very large simulations we will consume more memory, but that is probably worth it.

**Comment by glwagner** [comment]
We opted against this strategy for two reasons:

1. It increases preprocessing time
2. It may not be feasible for very large datasets, such as the JRA55 data set (1.7 T)

**Comment by simone-silvestri** [comment]
Right, lately I discovered that this strategy worked quite well for ECCO data. 
Given the problems that we are having lately we could think to reconsider it (we have to change JRA55 backend anyways to allow multi-year forcing). Note that the interpolation is done as a preprocessing step only when the memory is filled in case of partially in-memory datasets, not at the start of the simulation (we wouldn't want to use fully in-memory for large datasets anyway). So every tot there would be some interpolation when the new data is loaded in memory but this is hardly a performance hit. The only added memory cost would be the difference between the used grid and the JRA55 native grid.

**Comment by glwagner** [comment]
True but the JRA55 native grid is coarse while the used grids (up to 1/96 degree) may be very fine. That could be a factor of 25^2 = 625 in memory cost.

There also may be no advantage if interpolating on the fly is basically free.
